browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -87,8 +49,14 @@ bool ApplyPotentialReferrerBlock(std::shared_ptr<BraveRequestInfo> ctx) {
 int OnBeforeURLRequest_SiteHacksWork(const ResponseCallback& next_callback,
                                      std::shared_ptr<BraveRequestInfo> ctx) {
   ApplyPotentialReferrerBlock(ctx);
-  if (ctx->request_url.has_query()) {
-    ApplyPotentialQueryStringFilter(ctx);
+  if (ctx->request_url.has_query() && ctx->allow_brave_shields) {
```

Comment:
We can probably simplify check to just `if (ctx->allow_brave_shields)` since `MaybeApplyQueryStringFilter()` now has an early return for URLs without a query string.

Reply:
Done